syntaxes/hlasm.tmLanguage.json

```diff
@@ -25,43 +22,70 @@
 				]
 		},
 		"lineTooLong": {
-			"_description": "Assembler lines should only be 72 chars, at 73 the entire line should be in error",
+			"_description": "Assembler lines should only be 80 chars, at 81 the entire line should be in error",
 			"patterns": [
 				{
 					"name": "invalid.illegal.hlasm",
-					"match": "^.{73,}"  
-				}
-			]
-		},
-		"lineComment": {
-			"_description": "All statements that have a * in column 1",
-			"patterns": [
-				{
-					"name": "comment.line.double-slash.hlasm",
-					"match": "^\\*.*$"
+					"match": "^.{81,}"  
 				}
 			]
 		},
-		"continuation": {
-			"_description": "Continuation character that appears in column 72",
+		"parseAllLines": {
+			"_description": "Separate out code from continuation, sequence field, and the rest",
 			"patterns": [
 				{
-					"match": "^(.{71,71})(.)$",
+					"match": "^(.{0,71})(.)(.{0,8})(.*)$",
```

Comment:
I believe that this is where the problem lies for the last character errors. You are searching for characters 0-71, another character, the next 8 characters, then any additional characters.

I might take a stab on getting the line numbers correct with the continuation in a separate branch of this code base.

Reply:
with the 71,71 notation, any lines <71 characters weren't being passed to the "1" capture group process.  the second line was my attempt to parse short lines.

Reply:
I can confirm that this had something to do with the problem. I believe this is due to the regex trying to match the entire line.

`.{0,71}` says match between 0 and 71 characters and the `(.)` requires a character after the match.

The result meant for any string that was less than 72 characters, characters `0` to `n - 1` were matched in the `.{0,71}` group and the nth character was matched by the `(.)`.